environment.go

```diff
@@ -237,12 +242,31 @@ func (env environment) runCommand(ctx context.Context, cmd *exec.Cmd, timeout ti
 	}
 }
 
-func (env environment) execGoGet(ctx context.Context, modulePath, moduleVersion string) error {
+// execGoGet runs "go get -d -v" with the given module/version as an argument.
+// Also allows passing in a second module/version pair, meant to be the main
+// Caddy module/version we're building against; this will prevent the
+// plugin module from causing the Caddy version to upgrade, if the plugin
+// version requires a newer version of Caddy.
+func (env environment) execGoGet(ctx context.Context, modulePath, moduleVersion, caddyModulePath, caddyVersion string) error {
```

Comment:
Maybe we can add a link to the issue or this PR in the comment for this method. This is one of those complex/weird cases where context would be good to preserve.